Lib/test/test_fstring.py

```diff
@@ -427,703 +482,1103 @@ def test_ast_line_numbers_with_parentheses(self):
+        self.assertEqual(len(w.warnings), 1)
+        self.assertEqual(w.warnings[0].category, SyntaxWarning)
+
+    def test_fstring_backslash_prefix_raw(self):
+        self.assertEqual(f"\\", "\\")
+        self.assertEqual(f"\\\\", "\\\\")
+        self.assertEqual(rf"\\", r"\\")
+        self.assertEqual(rf"\\\\", r"\\\\")
+        self.assertEqual(rf"\\", r"\\")
+        self.assertEqual(rf"\\\\", r"\\\\")
+        self.assertEqual(Rf"\\", R"\\")
+        self.assertEqual(Rf"\\\\", R"\\\\")
+        self.assertEqual(Rf"\\", R"\\")
+        self.assertEqual(Rf"\\\\", R"\\\\")
+        self.assertEqual(Rf"\\", R"\\")
+        self.assertEqual(Rf"\\\\", R"\\\\")
+
+    def test_fstring_format_spec_greedy_matching(self):
+        self.assertEqual(f"{1:}}}", "1}")
+        self.assertEqual(f"{1:>3{5}}}}", "                                  1}")
 
     def test_yield(self):
         # Not terribly useful, but make sure the yield turns
         #  a function into a generator
         def fn(y):
-            f'y:{yield y*2}'
-            f'{yield}'
+            f"y:{yield y*2}"
+            f"{yield}"
 
         g = fn(4)
         self.assertEqual(next(g), 8)
         self.assertEqual(next(g), None)
 
     def test_yield_send(self):
         def fn(x):
-            yield f'x:{yield (lambda i: x * i)}'
+            yield f"x:{yield (lambda i: x * i)}"
 
         g = fn(10)
         the_lambda = next(g)
         self.assertEqual(the_lambda(4), 40)
-        self.assertEqual(g.send('string'), 'x:string')
+        self.assertEqual(g.send("string"), "x:string")
 
-    def test_expressions_with_triple_quoted_strings(self):
-        self.assertEqual(f"{'''x'''}", 'x')
-        # TODO: RUSTPYTHON self.assertEqual(f"{'''eric's'''}", "eric's")
+    # TODO: RUSTPYTHON SyntaxError
+    # def test_expressions_with_triple_quoted_strings(self):
+    #     self.assertEqual(f"{'''x'''}", 'x')
+    #     self.assertEqual(f"{'''eric's'''}", "eric's")
 
-        # Test concatenation within an expression
-        # TODO: RUSTPYTHON self.assertEqual(f'{"x" """eric"s""" "y"}', 'xeric"sy')
-        # TODO: RUSTPYTHON self.assertEqual(f'{"x" """eric"s"""}', 'xeric"s')
-        # TODO: RUSTPYTHON self.assertEqual(f'{"""eric"s""" "y"}', 'eric"sy')
-        # TODO: RUSTPYTHON self.assertEqual(f'{"""x""" """eric"s""" "y"}', 'xeric"sy')
-        # TODO: RUSTPYTHON self.assertEqual(f'{"""x""" """eric"s""" """y"""}', 'xeric"sy')
-        # TODO: RUSTPYTHON self.assertEqual(f'{r"""x""" """eric"s""" """y"""}', 'xeric"sy')
+    #     # Test concatenation within an expression
+    #     self.assertEqual(f'{"x" """eric"s""" "y"}', 'xeric"sy')
+    #     self.assertEqual(f'{"x" """eric"s"""}', 'xeric"s')
+    #     self.assertEqual(f'{"""eric"s""" "y"}', 'eric"sy')
+    #     self.assertEqual(f'{"""x""" """eric"s""" "y"}', 'xeric"sy')
+    #     self.assertEqual(f'{"""x""" """eric"s""" """y"""}', 'xeric"sy')
+    #     self.assertEqual(f'{r"""x""" """eric"s""" """y"""}', 'xeric"sy')
```

Comment:
I think this part is the reason why I commented line by line of broken assertion. 
Only one assertion can be run, and others make SyntaxError